static/gsApp/views/seerAutomation/components/repoProviderIcon.tsx

```diff
@@ -0,0 +1,27 @@
+import {IconBitbucket} from 'sentry/icons/iconBitbucket';
+import {IconGithub} from 'sentry/icons/iconGithub';
+import {IconGitlab} from 'sentry/icons/iconGitlab';
+import {IconOpen} from 'sentry/icons/iconOpen';
+import {IconVsts} from 'sentry/icons/iconVsts';
+import type {SVGIconProps} from 'sentry/icons/svgIcon';
+
+const PROVIDER_ICONS = {
+  github: IconGithub,
+  'integrations:github': IconGithub,
+  'integrations:github_enterprise': IconGithub,
+  bitbucket: IconBitbucket,
+  'integrations:bitbucket': IconBitbucket,
+  visualstudio: IconVsts,
+  'integrations:vsts': IconVsts,
+  gitlab: IconGitlab,
+  'integrations:gitlab': IconGitlab,
+};
+
+interface Props extends SVGIconProps {
+  provider: string;
+}
+
+export default function RepoProviderIcon({provider, ...props}: Props) {
+  const Icon = PROVIDER_ICONS[provider as keyof typeof PROVIDER_ICONS] ?? IconOpen;
```

Comment:
The default code path + a loose string type seems like a dangerous pattern. If the integration list changes, we'll hit the default without ever knowing. Wdyt about adding a log line here for visibility?